src/python/zquantum/core/wip/circuits/_circuit.py

```diff
@@ -1,9 +1,11 @@
 from functools import singledispatch, reduce
+import operator
 from typing import Union, Dict, Optional, Iterable, Any
 
 import sympy
 
-from . import _gates
+from . import _gates, MatrixFactoryGate
```

Comment:
You're importing `MatrixFactoryGate` from the `__init__` (I suppose PyCharm generated this import), did you consider importing the symbol from the module it was originally defined in? 

My gut feeling is that the latter approach would be more coherent with the scenarios where we import symbols not exposed in the public API (probably the majority of them), but I'm happy to hear your thoughts.

Reply:
I agree, and indeed, the import was autogenerated. I addressed this in 9e1bdf9.